File: Documentation/API/Parsers/Midi/ReadTempoChangesFromMidiData.cpp
#include <iostream>

#include "RhythmGameUtilities/File.hpp"
#include "RhythmGameUtilities/Parsers/Midi.hpp"

using namespace RhythmGameUtilities;

auto main() -> int
{
    auto bytes = ReadBytesFromFile("./song.mid");

    auto tempoChanges = ReadTempoChangesFromMidiData(bytes);

    std::cout << size(tempoChanges) << std::endl; // 7

    return 0;
}

File: Documentation/API/Parsers/Midi/ReadTempoChangesFromMidiData.js
import { readFileSync } from 'node:fs';

import RhythmGameUtilities from '@rhythm-game-utilities/core';

const data = readFileSync('./tests/Mocks/song.mid');

const tempoChanges = RhythmGameUtilities.ReadTempoChangesFromMidiData(data);

console.log(tempoChanges.length); // 1
